sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1222,6 +1222,11 @@ __SYCL_EXPORT pi_result piEnqueueNativeKernel(
     const void **args_mem_loc, pi_uint32 num_events_in_wait_list,
     const pi_event *event_wait_list, pi_event *event);
 
+__SYCL_EXPORT pi_result piEnqueueBarrierWait(pi_queue command_queue,
```

Comment:
This is only different from `piEnqueueEventsWait` in that it blocks subsequent commands execution in OOO queue, right? They are named similarly in OpenCL (clEnqueueMarkerWithWaitList & clEnqueueBarrierWithWaitList), so let's also keep these very similar PI interfaces named similarly, e.g. piEnqueueEventsWaitWithBarrier?

Reply:
Since the "level_zero" plugin is merged now, please support it there as well. This seems to be the exact match: https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zecommandlistappendbarrier#_CPPv426zeCommandListAppendBarrier24ze_command_list_handle_t17ze_event_handle_t8uint32_tP17ze_event_handle_t

Reply:
OK, sounds good. Will rename it. And also will add the implementation for Level Zero here, as it was merged yesterday after this PR had been created.

Reply:
Fixed.